subprojects/core/src/main/java/org/gradle/execution/taskpath/ProjectFinderByTaskPath.java

```diff
@@ -19,40 +19,123 @@
 import org.gradle.api.InvalidUserDataException;
 import org.gradle.api.Project;
 import org.gradle.api.internal.project.ProjectInternal;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
 import org.gradle.util.internal.NameMatcher;
 
 import java.util.Map;
 
 public class ProjectFinderByTaskPath {
+    private static final Logger LOGGER = Logging.getLogger(ProjectFinderByTaskPath.class);
 
-    public ProjectInternal findProject(String projectPath, ProjectInternal startFrom) {
```

Comment:
This appears to be a public API; is it OK to change it without warning?  I'm not sure if making it now throw an exception will break ABI compatibility.

Reply:
Extreme nitpick: add the new method `resolveProject` after this one; it makes the diff a bit prettier.

Reply:
Prove me wrong, but I believe `public ProjectInternal findProject(String projectPath, ProjectInternal startProject) throws ProjectLookupException` should be API/ABI compatible with the previous version:
- The second parameter name, as maybe it's more descriptive. By this, the ABI should be the same, as parameters are same typed, same numbered (IDK if name changes considered ABI breaking, maybe reflection usage could use troubles?). I've already met with binary compatibility issues, and this one in particular was not shown as an issue.
- The method is specialized with (even if RuntimeException) throws declaration, which should only document more clearly that this can throw an exception. I think this is OK, as the method has a stronger signature, so it can substitute the throw-free version.

Reply:
Sorry about the delay - if it works and passes our automated ABI checks, go for it!

Reply:
:+1: